lib/Model/Query.js

```diff
@@ -293,7 +293,12 @@ Query.prototype._shareSubscribe = function(options, cb) {
 Query.prototype._removeMapIds = function(ids) {
   for (var i = ids.length; i--;) {
     var id = ids[i];
-    delete this.idMap[id];
+    if (this.idMap[id] > 0) {
+      this.idMap[id]--;
+    }
+    if (this.idMap[id] === 0) {
+      delete this.idMap[id];
+    }
```

Comment:
Ok, this is done now too, thanks for the suggestion! It's one fewer conditional check too, which is nice.